charts/pact-broker/templates/serviceaccount.yaml

```diff
@@ -1,19 +1,27 @@
 {{- if .Values.serviceAccount.create }}
+{{- with .Values.serviceAccount }}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ .Values.serviceAccount.name }}
-  namespace: {{ .Release.Namespace }}
+  name: {{ .name}}
```

Comment:
Not sure this will work here? `.name` as far as I can see doesn't render to any value, and it also means that the service account used in the deployment won't match the one that is created here.

Reply:
Yes it does. By using with .Values.serviceName earlier it means we don't have to repeat the full path over and over. It renders locally, you can test it.

Reply:
When using `with` like I do here: https://github.com/pact-foundation/pact-broker-chart/blob/ab1ebe6d2a463a0f01c648c55f81966c62779af0/charts/pact-broker/templates/serviceaccount.yaml#L2 it allows you to enter that context and reference everything from that key level. This is outlined: https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with but yeah.